tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1114 @@
+
+	var trueFlag = true
+
+	userUID := new(int64)
+
+	*userUID = 0
+
+	securityContext := &corev1.SecurityContext{
+		RunAsUser:  userUID,
+		Privileged: &trueFlag,
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"NET_RAW", "NET_ADMIN", "SYS_ADMIN", "IPC_LOCK"},
+		},
+	}
+
+	glog.V(100).Infof("Setting SecurityContext")
+
+	deploymentContainer = deploymentContainer.WithSecurityContext(securityContext)
+
+	glog.V(100).Infof("Dropping ALL security capability")
+
+	deploymentContainer = deploymentContainer.WithDropSecurityCapabilities([]string{"ALL"}, true)
+
+	glog.V(100).Infof("%q container's  definition:\n%#v", cName, deploymentContainer)
+
+	return deploymentContainer
+}
+
+func definePodLevelBondTestPodDeployment(
+	apiClient *clients.Settings,
+	containerConfig *corev1.Container,
+	deploymentName,
+	nsName,
+	scheduleOnHost,
+	sriovNet1Name,
+	sriovNet2Name,
+	bondNetName,
+	bondInfIPv4,
+	bondInfIPv6,
+	bondInfSubMaskIPv4,
+	bondInfSubMaskIPv6,
+	bondInfMacAddr string,
+	deployLabels map[string]string) *deployment.Builder {
+	glog.V(100).Infof("Defining deployment %q in %q ns", deploymentName, nsName)
+
+	nodeSelector := map[string]string{"kubernetes.io/hostname": scheduleOnHost}
+
+	netAnnotations := []*types.NetworkSelectionElement{
+		{
+			Name:      sriovNet1Name,
+			Namespace: nsName,
+		},
+		{
+			Name:      sriovNet2Name,
+			Namespace: nsName,
+		},
+		{
+			Name:      bondNetName,
+			Namespace: nsName,
+			IPRequest: []string{fmt.Sprintf("%s/%s", bondInfIPv4, bondInfSubMaskIPv4),
+				fmt.Sprintf("%s/%s", bondInfIPv6, bondInfSubMaskIPv6)},
+			MacRequest: bondInfMacAddr,
+		},
+	}
+
+	podDeployment := deployment.NewBuilder(apiClient, deploymentName, nsName, deployLabels, *containerConfig)
+
+	glog.V(100).Infof("Assigning ServiceAccount %q to the deployment", podLevelBondDeploymentSAName)
+
+	podDeployment = podDeployment.WithServiceAccountName(podLevelBondDeploymentSAName)
+
+	podDeployment = podDeployment.WithNodeSelector(nodeSelector)
+
+	podDeployment = podDeployment.WithSecondaryNetwork(netAnnotations)
+
+	return podDeployment
+}
+
+func generateTCPTraffic(clientPod *pod.Builder, serverIPAddr, serverPort string) (string, error) {
+	By("Generating TCP traffic")
+
+	glog.V(100).Infof("Ensure pod %q in namespace %q is Ready",
+		clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+	err := clientPod.WaitUntilReady(5 * time.Second)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to wait for pod %q in namespace %q to become Ready: %v",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+		return "", fmt.Errorf("failed to wait for pod %q in namespace %q to become Ready: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	cmdToRun := []string{"bash", "-c",
+		fmt.Sprintf("testcmd -protocol tcp -port %s -interface net3 -server %s",
```

Comment:
do U think it worth parameterizing protocol ?

Reply:
I don't think so, for what? this is internal method and we have to use reliable protocol. for the testcmd tool we have only 4 options: icmp, udp, tcp or sctp. icmp&udp not reliable, sctp not required in our case